### DISCUSSION TITLE
tressitter parser timeout and range query

### DISCUSSION THREAD

**Discussion by cathaysia** [neovim/neovim#24302] [Q&A] [answered]
Currently, if there is a problem with the treesitter's parser (such as an infinite loop), it will cause nvim to freeze and fail to accept any commands. The only way to do this is to kill neovim with kill. Whether neovim should set a timeout condition.

Also, neovim performs very poorly with large files (these "large files" are only around 20MB). Disabling the treesitter provided a noticeable improvement. I observed that helix only renders in viewable area (eg inlayhints and treesitter). Is it possible to integrate such functionality in neovim?

**Comment by clason** [comment, marked as answer]
https://github.com/neovim/neovim/issues/22426

Help is welcome.
